js/yasgui.local.js

```diff
@@ -1544,26 +1547,53 @@ function refreshList(){
 
 function shouldShowExample(elem){
   var search = document.getElementById('searchExamples').value;
+  var tagsSelected = getSelectedTags();
+  
+  //pas de filtre
+  if(tagsSelected.length === 0 && search === "") { return true; }
 
-  if(search === "") { return true; }
-
-  if(elem.title !== undefined && elem.title.includes(search, 0)) return true;
-  if(elem.description !== undefined && elem.description.includes(search, 0)) return true;
-
+  //filtre uniquement sur titre et description
+  if(tagsSelected.length === 0) {
+    if(elem.title !== undefined && elem.title.includes(search, 0)) return true;
+    if(elem.description !== undefined && elem.description.includes(search, 0)) return true; 
+  }
+  
+  //filtre si tags selectionnes
+  if(elem.tags === undefined)  { return false; }
+  for(var i = 0; i < elem.tags.length; i++){
```

Comment:
Je pense qu'on aurait pu se passer du `for`, et se simplifier la vie avec une méthode de `Array` du genre [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) ou `any`.

```js
const matchTitle = e => e.title !== undefined && e.title.includes(search);
const matchDescription = e => e.description !== undefined && e.description.includes(search);

return elem.tags.some(matchTitle) || elem.tags.some(matchDescription);
```

Au passage, on peut réutiliser les fonctions `match*` dans les lignes précédentes.

Et d'ailleurs, on aurait pu aussi écrire une fonction `matchTitleOrDescription`:

```js
const matchTitleOrDescription = e => matchTitle(e) || matchDescription(e);
```

 ce qui éviterait sans doute de faire plusieurs passages dans le tableau avec `some`:

```js
const matchTitle = e => e.title !== undefined && e.title.includes(search);
const matchDescription = e => e.description !== undefined && e.description.includes(search);
const matchTitleOrDescription = e => matchTitle(e) || matchDescription(e);

return elem.tags.some(matchTitleOrDescription);
```

Oups. J'ai oublié le plus important: la sélection des tags... Bon, je ne vais pas tout écrire, mais tu vois l'idée?